paperqa/agents/models.py

```diff
@@ -0,0 +1,535 @@
+        "The current status of evidence/papers/cost is {status}"
+    )
+    paper_directory: str | os.PathLike = Field(
+        default=Path("."),
+        description=(
+            "Local directory which contains the papers to be indexed and searched."
+        ),
+    )
+    index_directory: str | os.PathLike | None = Field(
+        default=None,
+        description=(
+            "Directory to store the PQA generated search index, configuration, and answer indexes."
+        ),
+    )
+    manifest_file: str | os.PathLike | None = Field(
+        default=None,
+        description=(
+            "Optional manifest CSV, containing columns which are attributes for a DocDetails object. "
+            "Only 'file_location','doi', and 'title' will be used when indexing."
+        ),
+    )
+    search_count: int = 8
+    wipe_context_on_answer_failure: bool = True
+    timeout: float = Field(
+        default=500.0,
+        description=(
+            "Matches LangChain AgentExecutor.max_execution_time (seconds), the timeout"
+            " on agent execution."
+        ),
+    )
+    should_pre_search: bool = Field(
+        default=False,
+        description="If set to true, run the search tool before invoking agent.",
+    )
+    agent_config: dict[str, Any] | None = Field(
+        default=None,
+        description="Optional keyword argument configuration for the agent.",
+    )
+    tool_names: Collection[str] | None = Field(
+        default=None,
+        description=(
+            "Optional override on the tools to provide the agent. Leaving as the"
+            " default of None will use a minimal toolset of the paper search, gather"
+            " evidence, collect cited papers from evidence, and gen answer. If passing tool"
+            " names (non-default route), at least the gen answer tool must be supplied."
+        ),
+    )
+
+    @field_validator("tool_names")
+    @classmethod
+    def validate_tool_names(cls, v: set[str] | None) -> set[str] | None:
+        if v is None:
+            return None
+        from .main import GenerateAnswerTool
+
+        answer_tool_name = GenerateAnswerTool.__fields__["name"].default
+        if answer_tool_name not in v:
+            raise ValueError(
+                f"If using an override, must contain at least the {answer_tool_name}."
+            )
+        return v
+
+
+class ParsingOptions(str, Enum):
+    PAPERQA_DEFAULT = "paperqa_default"
+
+    def available_for_inference(self) -> list[ParsingOptions]:
+        return [self.PAPERQA_DEFAULT]  # type: ignore[list-item]
+
+    def get_parse_type(self, config: ParsingConfiguration) -> str:
+        if self == ParsingOptions.PAPERQA_DEFAULT:
+            return config.parser_version_string
+        assert_never()
+
+
+class ChunkingOptions(str, Enum):
+    SIMPLE_OVERLAP = "simple_overlap"
+
+    # Note that SIMPLE_OVERLAP must be valid for all by default
```

Comment:
This comment seems misplaced, can you move it next to `SIMPLE_OVERLAP`?

Reply:
should go in the func below this actually.